py_ppo.py

```diff
@@ -0,0 +1,181 @@
+         ('cones absent', 'obo:PPO_0002645')]
+    """
+    if present is None:
+        url = f"https://biscicol.org/api/v2/ppo/all_short"
+    elif present is True:
+        url = f"https://biscicol.org/api/v2/ppo/present_short"
+    elif present is False:
+        url = f"https://biscicol.org/api/v2/ppo/absent_short"
+    else:
+        raise ValueError(f"Present should be None or Boolean, got {present}")
+
+    r = requests.get(url)
+
+    if r.status_code == 200:
+        return r.json()
+
+    raise InvalidRequestError(f"Request failed with status code {r.status_code}")
+
+
+def download(explode=False, limit=100, timeout=3.05, debug=False, **options):
+    """Download data from the plant phenology data portal.
+
+    This function builds a query string from the provided arguments, and uses it
+    to fetch data from the REST API into a temporary folder, then unpacks the
+    data and loads it into a pandas dataframe.
+
+    Args:
+        limit: Maximum number of records to retreive
+        explode: If True, each termID will get its own row
+        timeout: Number of seconds to wait for the server to respond
+        **options: keyword arguments used to filter the data before retreiving
+            it from the server.
+
+    Notes:
+        Similar to
+        [rppo.ppo_data()](https://docs.ropensci.org/rppo/articles/rppo-vignette.html#ppo_data-function)
+
+        The queries build here follow lucene query syntax; see
+        https://www.elastic.co/guide/en/kibana/current/lucene-query.html. They
+        may look something like this:
+        https://biscicol.org/api/v3/download/_search?limit=5&q=genus:Syringa+AND+year:[2000
+        TO 2021]
+
+    """
+    url = _build_url(limit, **options)
+
+    if debug:
+        print(f"Trying to fetch data from {url}")
+
+    # Wait 3 seconds for a connection, 30 for the response.
+    response = requests.get(url, timeout=timeout)
+
+    if response.status_code == 200:
+        return _to_dataframe(response, explode)
+
+    if response.status_code == 204:
+        print("No data found, you may try to broaden your search.")
+        return pd.DataFrame()  # Trying to be consistent in what we return
+
+    raise InvalidRequestError(f"Requests failed with status code {response.status_code}. Please raise an issue.")
+
+
+def _build_url(limit, **options):
+    """Parse options to build query string."""
+    base_url = f"https://biscicol.org/api/v3/download/_search?limit={limit}"
+
+    if "termID" in options:
+        # These need special formatting to retain quotes in URL, otherwise the
+        # colon in "obo:PPO_XXXXXXX" messes things up.
+        options['termID'] = f"\"{options['termID']}\""
+
+    # Build query string
+    query = "&q=" + "+AND+".join([f"{k}:{v}" for k, v in options.items()])
+
+    return base_url + query
+
+
+def _to_dataframe(response, explode):
+    """Extract response body and load as pandas dataframe."""
+
+    with tempfile.TemporaryDirectory() as tempdir:
```

Comment:
why not pass a `downlod_dir` instead of using `TemporaryDirectory` as it generates random temp directories? I found that those random dirs are not accessible to users. As an user I want to check the content of the downloaded files and `data.csv`.

Reply:
Actually I didn't want the user to see that there are downloaded files and directories at all. I want to pretend the data came directly from the server. Perhaps with an updated output message you don't feel like you're missing out anymore? But see #7 for follow-up.